.github/workflows/pull_request.yml

```diff
@@ -0,0 +1,83 @@
+# Copyright 2018-2020 The OpenEBS Authors. All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+name: ci
+
+on:
+  pull_request:
+    branches:
+      # on pull requests to master and release branches
+      - master
+      - 'v*'
+
+jobs:
+  lint:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v2
+
+      - name: Shellcheck
+        uses: reviewdog/action-shellcheck@v1
+        with:
+          github_token: ${{ secrets.github_token }}
+          reporter: github-pr-review
+          path: '.'
+          pattern: '*.sh'
+
+  cstor-base:
+    runs-on: ubuntu-latest
+    needs: ['lint']
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v2
+
+      - name: Set up QEMU
+        uses: docker/setup-qemu-action@v1
+        with:
+          platforms: all
+
+      - name: Set up Docker Buildx
+        id: buildx
+        uses: docker/setup-buildx-action@v1
+        with:
+          version: latest
+
+      - name: Build Image
+        env:
+          IMG_RESULT: cache
```

Comment:
If you cache the image, you wont be able to use the image. after exiting buildix. It exists only in the buildkit cache.

For pull requests, [this line](https://github.com/openebs/libcstor/pull/71/files?file-filters%5B%5D=.Dockerfile&file-filters%5B%5D=.yml#diff-6e18312c20a206d6e39bc46571987be2cd11a3f8cdd46ffdf7a46dc04dcaf067R21) will always pull image from the docker hub. It wont use the locally built cstor-base image.

We may need to push the image to an intermediate repo to use that image.

Reply:
IMO we can just remove the cstor build and have only the cstor-base build in pull request as the cstor image only copies a entrypoint script to the cstor-base image.

Reply:
Ok, Then I think for now, that would work. In the long term, will need to identify a way to push intermediate images and use them.